src/assets/wise5/vle/timed-node/timed-node.component.ts

```diff
@@ -0,0 +1,191 @@
+import { ComponentStateInfoComponent } from '../../common/component-state-info/component-state-info.component';
+import { ConfigService } from '../../services/configService';
+import { ConstraintService } from '../../services/constraintService';
+import { FlexLayoutModule } from '@angular/flex-layout';
+import { HelpIconComponent } from '../../themes/default/themeComponents/helpIcon/help-icon.component';
+import { MatButtonModule } from '@angular/material/button';
+import { MatCardModule } from '@angular/material/card';
+import { MatDividerModule } from '@angular/material/divider';
+import { NodeComponent } from '../node/node.component';
+import { NodeStatusService } from '../../services/nodeStatusService';
+import { SessionService } from '../../services/sessionService';
+import { StudentDataService } from '../../services/studentDataService';
+import { StudentNodeService } from '../../services/studentNodeService';
+import { SubmitSurveyComponent } from '../submit-survey/submit-survey.component';
+import { TimedNodeService } from '../../services/timedNodeService';
+import { VLEProjectService } from '../vleProjectService';
+
+@Component({
+  imports: [
+    CommonModule,
+    ComponentComponent,
+    ComponentStateInfoComponent,
+    FlexLayoutModule,
+    HelpIconComponent,
+    MatButtonModule,
+    MatCardModule,
+    MatDividerModule,
+    SubmitSurveyComponent
+  ],
+  selector: 'timed-node',
+  styleUrl: './timed-node.component.scss',
+  templateUrl: './timed-node.component.html'
+})
+export class TimedNodeComponent extends NodeComponent {
+  private componentTimers: number[];
+  private currentComponentIndex = 0;
+  private currentInterval: any;
+  protected stepCompleted = false;
+  protected timerCountDown: number;
+
+  constructor(
+    protected componentService: ComponentService,
+    protected configService: ConfigService,
+    protected constraintService: ConstraintService,
+    protected nodeService: StudentNodeService,
+    protected nodeStatusService: NodeStatusService,
+    protected projectService: VLEProjectService,
+    protected sessionService: SessionService,
+    protected studentDataService: StudentDataService,
+    private timedNodeService: TimedNodeService
+  ) {
+    super(
+      componentService,
+      configService,
+      constraintService,
+      nodeService,
+      nodeStatusService,
+      projectService,
+      sessionService,
+      studentDataService
+    );
+  }
+
+  async ngOnInit(): Promise<void> {
+    super.ngOnInit();
+    await this.skipViewedComponents();
+    this.timedNodeService.broadcastIsNodeCompleted(this.stepCompleted);
+    if (!this.stepCompleted) {
+      this.componentTimers = this.node.components.map((component: ComponentContent) => {
+        return component.timeLimit ?? 0;
+      });
+      this.components.forEach((component, index) => {
+        this.componentToVisible[component.id] = index === this.currentComponentIndex;
+      });
```

Comment:
Can we call this.hideComponents() here?
```suggestion
      this.components.forEach((component, index) => {
        this.componentToVisible[component.id] = index === this.currentComponentIndex;
      });
```